File: tests/abi/speaker_core_layout.cpp
#include "pluginterfaces/vst/vstspeaker.h"
#include <cstring>
#include "public.sdk/source/vst/vstspeakerarray.h"

#include <cstdio>

int main ()
{
	std::printf ("kSpeakerL %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerL));
	std::printf ("kSpeakerR %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerR));
	std::printf ("kSpeakerLfe %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerLfe));
	std::printf ("kSpeakerTfl %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerTfl));
	std::printf ("kSpeakerBfl %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerBfl));
	std::printf ("kSpeakerACN24 %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerACN24));
	std::printf ("kSpeakerRw %llu\n", static_cast<unsigned long long> (Steinberg::Vst::kSpeakerRw));
	std::printf ("kStereo %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::kStereo));
	std::printf ("k51 %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::k51));
	std::printf ("k71Music %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::k71Music));
	std::printf ("kAmbi1stOrderACN %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::kAmbi1stOrderACN));
	std::printf ("kAmbi7thOrderACN %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::kAmbi7thOrderACN));
	std::printf ("k50_4 %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::k50_4));
	std::printf ("k71_4 %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::k71_4));
	std::printf ("getChannelCount.k51 %d\n", Steinberg::Vst::SpeakerArr::getChannelCount (Steinberg::Vst::SpeakerArr::k51));
	std::printf ("getSpeakerIndex.R.stereo %d\n", Steinberg::Vst::SpeakerArr::getSpeakerIndex (Steinberg::Vst::kSpeakerR, Steinberg::Vst::SpeakerArr::kStereo));
	std::printf ("getSpeaker.k51.2 %llu\n", static_cast<unsigned long long> (Steinberg::Vst::SpeakerArr::getSpeaker (Steinberg::Vst::SpeakerArr::k51, 2)));
	std::printf ("isSubsetOf.stereo.51 %d\n", Steinberg::Vst::SpeakerArr::isSubsetOf (Steinberg::Vst::SpeakerArr::kStereo, Steinberg::Vst::SpeakerArr::k51));
	std::printf ("hasTopSpeakers.50_4 %d\n", Steinberg::Vst::SpeakerArr::hasTopSpeakers (Steinberg::Vst::SpeakerArr::k50_4));
	std::printf ("hasBottomSpeakers.50_4 %d\n", Steinberg::Vst::SpeakerArr::hasBottomSpeakers (Steinberg::Vst::SpeakerArr::k50_4));
	std::printf ("hasMiddleSpeakers.50_4 %d\n", Steinberg::Vst::SpeakerArr::hasMiddleSpeakers (Steinberg::Vst::SpeakerArr::k50_4));
	std::printf ("hasLfe.51 %d\n", Steinberg::Vst::SpeakerArr::hasLfe (Steinberg::Vst::SpeakerArr::k51));
	std::printf ("is3D.50_4 %d\n", Steinberg::Vst::SpeakerArr::is3D (Steinberg::Vst::SpeakerArr::k50_4));
	std::printf ("isAmbisonics.ambi1 %d\n", Steinberg::Vst::SpeakerArr::isAmbisonics (Steinberg::Vst::SpeakerArr::kAmbi1stOrderACN));
	Steinberg::Vst::SpeakerArray speakerArray (Steinberg::Vst::SpeakerArr::k51);
	std::printf ("SpeakerArray.total.51 %d\n", speakerArray.total ());
	std::printf ("SpeakerArray.at.51.0 %llu\n", static_cast<unsigned long long> (speakerArray.at (0)));
	std::printf ("SpeakerArray.at.51.3 %llu\n", static_cast<unsigned long long> (speakerArray.at (3)));
	std::printf ("SpeakerArray.getArrangement.51 %llu\n", static_cast<unsigned long long> (speakerArray.getArrangement ()));
	std::printf ("SpeakerArray.getSpeakerIndex.Lfe %d\n", speakerArray.getSpeakerIndex (Steinberg::Vst::kSpeakerLfe));
	speakerArray.setArrangement (Steinberg::Vst::SpeakerArr::k50_4);
	std::printf ("SpeakerArray.total.50_4 %d\n", speakerArray.total ());
	std::printf ("SpeakerArray.getArrangement.50_4 %llu\n", static_cast<unsigned long long> (speakerArray.getArrangement ()));
	return 0;
}

File: tools/speaker_core_layout.zig
const std = @import("std");
const speaker = @import("vst3-zig").pluginterfaces.vst.vstspeaker;

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try stdout.print("kSpeakerL {}\n", .{speaker.kSpeakerL});
    try stdout.print("kSpeakerR {}\n", .{speaker.kSpeakerR});
    try stdout.print("kSpeakerLfe {}\n", .{speaker.kSpeakerLfe});
    try stdout.print("kSpeakerTfl {}\n", .{speaker.kSpeakerTfl});
    try stdout.print("kSpeakerBfl {}\n", .{speaker.kSpeakerBfl});
    try stdout.print("kSpeakerACN24 {}\n", .{speaker.kSpeakerACN24});
    try stdout.print("kSpeakerRw {}\n", .{speaker.kSpeakerRw});
    try stdout.print("kStereo {}\n", .{speaker.SpeakerArr.kStereo});
    try stdout.print("k51 {}\n", .{speaker.SpeakerArr.k51});
    try stdout.print("k71Music {}\n", .{speaker.SpeakerArr.k71Music});
    try stdout.print("kAmbi1stOrderACN {}\n", .{speaker.SpeakerArr.kAmbi1stOrderACN});
    try stdout.print("kAmbi7thOrderACN {}\n", .{speaker.SpeakerArr.kAmbi7thOrderACN});
    try stdout.print("k50_4 {}\n", .{speaker.SpeakerArr.k50_4});
    try stdout.print("k71_4 {}\n", .{speaker.SpeakerArr.k71_4});
    try stdout.print("getChannelCount.k51 {}\n", .{speaker.getChannelCount(speaker.SpeakerArr.k51)});
    try stdout.print("getSpeakerIndex.R.stereo {}\n", .{speaker.getSpeakerIndex(speaker.kSpeakerR, speaker.SpeakerArr.kStereo)});
    try stdout.print("getSpeaker.k51.2 {}\n", .{speaker.getSpeaker(speaker.SpeakerArr.k51, 2)});
    try stdout.print("isSubsetOf.stereo.51 {}\n", .{@intFromBool(speaker.isSubsetOf(speaker.SpeakerArr.kStereo, speaker.SpeakerArr.k51))});
    try stdout.print("hasTopSpeakers.50_4 {}\n", .{@intFromBool(speaker.hasTopSpeakers(speaker.SpeakerArr.k50_4))});
    try stdout.print("hasBottomSpeakers.50_4 {}\n", .{@intFromBool(speaker.hasBottomSpeakers(speaker.SpeakerArr.k50_4))});
    try stdout.print("hasMiddleSpeakers.50_4 {}\n", .{@intFromBool(speaker.hasMiddleSpeakers(speaker.SpeakerArr.k50_4))});
    try stdout.print("hasLfe.51 {}\n", .{@intFromBool(speaker.hasLfe(speaker.SpeakerArr.k51))});
    try stdout.print("is3D.50_4 {}\n", .{@intFromBool(speaker.is3D(speaker.SpeakerArr.k50_4))});
    try stdout.print("isAmbisonics.ambi1 {}\n", .{@intFromBool(speaker.isAmbisonics(speaker.SpeakerArr.kAmbi1stOrderACN))});
    var speaker_array = speaker.SpeakerArray.init(speaker.SpeakerArr.k51);
    try stdout.print("SpeakerArray.total.51 {}\n", .{speaker_array.total()});
    try stdout.print("SpeakerArray.at.51.0 {}\n", .{speaker_array.at(0)});
    try stdout.print("SpeakerArray.at.51.3 {}\n", .{speaker_array.at(3)});
    try stdout.print("SpeakerArray.getArrangement.51 {}\n", .{speaker_array.getArrangement()});
    try stdout.print("SpeakerArray.getSpeakerIndex.Lfe {}\n", .{speaker_array.getSpeakerIndex(speaker.kSpeakerLfe)});
    speaker_array.setArrangement(speaker.SpeakerArr.k50_4);
    try stdout.print("SpeakerArray.total.50_4 {}\n", .{speaker_array.total()});
    try stdout.print("SpeakerArray.getArrangement.50_4 {}\n", .{speaker_array.getArrangement()});
}
